wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -499,6 +499,12 @@ public final class ZMUserSession: NSObject {
         restoreDebugCommandsState()
         configureRecurringActions()
 
+        // Proactively keep the self user in sync, which helps add resilience
+        // in cases where the self client may otherwise only have limited
+        // one time opportunities to discover important changes.
+        let selfUser = ZMUser.selfUser(in: managedObjectContext)
+        selfUser.needsToBeUpdatedFromBackend = true
```

Comment:
Historically clients could only discover the team during a slow sync after log in, now that a user can be associated with a team after login we need to be more proactive about discovering the team, otherwise some clients may miss their opportunity to discover the team because, for example, they were an old client that doesn't process team creation events.

With this change, we fetch the self user  and  transitively the team on launch, which in most cases may be unnecessary, but is a small cost for extra protection.